client/components/addAttendance/addAttendance.directive.js

```diff
@@ -25,6 +31,52 @@ angular.module('observatory3App')
     $scope.formTypeEmptyError = !$scope.attend.type;
   };
 
+  function getAttendanceStatus(data) {
+    var date = data.date,
+    mode = data.mode;
+    if (mode === 'day') {
+      var dayToCheck = new Date(date).toDateString();
+      for(var i = 0; i< $scope.user.attendance.length; i++){
+        var currentDay = new Date($scope.user.attendance[i].date).toDateString();
+        var isVerified = $scope.user.attendance[i].verified;
+        if(dayToCheck ===currentDay){
+          if(isVerified){
+            return 'attended';
+          } else {
+            return 'unverified';
+          }
+        }
+      }
+    }
+    return '';
+  }
+
+  function fixOffByOneDay(daystring){
+    var OffByoneday = daystring.split('T');
+    var correctDay = OffByoneday[0] + "T04:00:00.000Z";
+    return correctDay;
+  }
+
+  function fixUserAttendance(user){
+    for(var i = 0; i < user.attendance.length;i++){
+      user.attendance[i].date = fixOffByOneDay(user.attendance[i].date);
+    }
+  }
+
+  function AttendedDay(){
```

Comment:
The value of user.attendance.date from the back end has the form of yyyy-mm-ddT00:00:00Z. When you pass that in new Date(...) it would give you a Date Object of one day before, so messed up the date comparison. Changed it to T04:00:00Z fixed it.